boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,146 @@
+ */
+package org.ow2.proactive.boot.microservices;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.*;
+
+import org.apache.log4j.Logger;
+
+
+public enum IAMStarter {
+
+    //singleton instance of IAMStarter
+    INSTANCE;
+
+    private static final Logger LOGGER = Logger.getLogger(IAMStarter.class);
+
+    private static final String SEPARATOR = File.separator;
+
+    private static final String MICROSERVICE_NAME = "iam.war";
+
+    private static final String READY_MARKER = "Ready to process requests";
+
+    private static final int TIMEOUT = 180;
+
+    private static final String[] JVM_ARGS = { "-Xmx2048M" };
+
+    private static Process process;
+
+    private static List<String> command = new ArrayList<>();
+
+    private static boolean started = false;
+
+    /**
+     *  Start IAM microservice
+     */
+    public static Process start(String paHome, String microservicesPath)
+            throws InterruptedException, IOException, ExecutionException {
+
+        if (!started) {
+            LOGGER.info("Starting IAM microservice...");
+
+            buildJavaCommand(paHome);
+            buildMicroservicePath(paHome, microservicesPath);
+
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+
+            process = processBuilder.start();
+            LOGGER.info(streamOutput(process.getInputStream()));
+
+            started = true;
+        }
+        return process;
+    }
+
+    /**
+     * Stream microservice output (using an executor) to check that it starts properly
+     */
+    private static String streamOutput(InputStream inputStream) throws InterruptedException, ExecutionException {
+
+        // Stream microservice output
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<String> future = executor.submit((Callable) () -> {
+
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+                String line;
+
+                while ((line = br.readLine()) != null) {
+                    if (line.contains(READY_MARKER)) {
+                        break;
+                    }
+                }
+            }
+            return "IAM Microservice started";
+        });
+
+        // Check for timeout
+        try {
+            return future.get(TIMEOUT, TimeUnit.SECONDS);
+        } catch (TimeoutException toe) {
+            return "IAM Microservice timed out to start. See the logs for the details.";
+        } finally {
+            executor.shutdownNow();
+            future.cancel(true);
+        }
+    }
+
+    /**
+     * Prepare java command with jvm args
+     */
+    private static void buildJavaCommand(String paHome) {
+
+        String javaCmd = paHome + SEPARATOR + "jre" + SEPARATOR + "bin" + SEPARATOR + "java";
+
+        if (new File(javaCmd).exists()) {
```

Comment:
this is not portable.
On Windows, new File(javaCmd).exists() will always return false.

To fix this, simply check the operating system, and build the javaCmd on Windows as
javaCmd = paHome + SEPARATOR + "jre" + SEPARATOR + "bin" + SEPARATOR + "java.exe"